src/MapViewController.swift

```diff
@@ -417,11 +476,184 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
     tgViewController.markerRemove(marker)
     return
   }
+}
 
-  //MARK: - TGRecognizerDelegate
-
+extension MapViewController : TGMapViewDelegate, TGRecognizerDelegate {
+  
+  //MARK : TGMapViewDelegate
+  
+  open func mapView(_ mapView: TGMapViewController, didLoadSceneAsync scene: String) {
+    guard (loadDelegate != nil) && (self.loadDelegate?.responds(to: #selector(MapLoadDelegate.mapController(_:didLoadSceneAsync:))))! else {
```

Comment:
Eek- I should have known better...!

For the optional methods we will still need a `respondsTo` check. Maybe I update `MapSelectDelegate` and `MapLoadDelegate` into even smaller protocols that don't have any optional methods? How do you feel about that?